lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -159,6 +185,17 @@ def add_payment(xml, card_or_token, options)
         end
       end
 
+      def add_check_payment(xml, check, options)
+        xml.hps :CheckAction, 'SALE'
+        xml.hps :AccountInfo do
+          xml.hps :RoutingNumber, check.routing_number
+          xml.hps :AccountNumber, check.account_number
+          xml.hps :CheckNumber, check.number
+          xml.hps :AccountType, check.account_type.upcase
+        end
+        xml.hps :CheckType, check.account_holder_type.upcase
```

Comment:
These fields are all required by HPS; attempting a payment without them will result in a gateway processing error.